pennylane/transforms/resolve_dynamic_wires.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains a transform for resolving dynamic wires into real wires.
+"""
+from collections.abc import Hashable, Sequence
+from typing import Optional
+
+from pennylane.allocation import Allocate, Deallocate
+from pennylane.measurements import measure
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.typing import PostprocessingFn, Result, ResultBatch
+
+from .core import transform
+
+
+class _WireManager:
+
+    def __init__(self, zeroed=(), dirty=(), min_integer=None):
+        self._zeroed = list(zeroed)
+        self._dirty = list(dirty)
+        self._loaned = {}  # wire to final register type
+        self.min_integer = min_integer
+
+    def get_wire(self, require_zeros):
+        """Retrieve a concrete wire label from available registers."""
+        if not self._zeroed and not self._dirty:
+            if self.min_integer is None:
+                raise ValueError("no wires left to allocate.")
+            self._zeroed.append(self.min_integer)
+            self.min_integer += 1
+        if require_zeros:
+            if self._zeroed:
+                w = self._zeroed.pop()
+                self._loaned[w] = "zeroed"
+                return w, []
+            w = self._dirty.pop()
+            self._loaned[w] = "dirty"
+            m = measure(w, reset=True)
```

Comment:
So dirty means that a qubit is in an unentangled state, right?
Or can it be entangled with other dirty qubits? :thinking: 
Also, dirty means that we do not need the dirty state any longer? (I suppose if we did, this would be in the scenario of borrowing with is out of scope for this PR :) )

Reply:
Hm. That still could include states that are entangled with algorithmic qubits, no?
Maybe actually `garbage` :sweat_smile:

Reply:
Given we reset them, we remove any entanglement before reuse.